### PR TITLE
First pass rootless tutorial

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
First pass of the rootless tutorial.  This may be come the basis
for a User Guide of the same.

Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Ugh, love the whitespace checks.  Will clean them up tomorrow, happy to take comments prior.

**Comment by rhatdan** [comment]
I think it is always easier to do the first pass at something like this in google docs.

File: docs/tutorials/rootless_tutorial.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+johndoe:100000:65536
+test:165536:65536
+```
+
+The format of this file is USERNAME:UID:RANGE
+
+* username as listed in /etc/passwd or getpwent.
+* The initial uid allocated for the user.
+* The size of the range of UIDs allocated for the user.
+
```
Comment (rhatdan) [inline docs/tutorials/rootless_tutorial.md]: Should add a NOTE stating that this is not currently supported with network installs. IE You have to have files like this locally, no way to set this up with LDAP or ActiveDirectory.

**Comment by rhatdan** [comment]
Also someone should figure out how one would go about setting up and installing podman without root privilege.  IE No admin capabilities at all.
Could I build podman, slirp4netns, fuse-overlayfs and install them in ~/bin, and then get storage.conf and libpod.conf to use them and get them working in my homedir.

File: docs/tutorials/rootless_tutorial.md (hunk 1 above)
Comment (TomSweeneyRedHat) [reply docs/tutorials/rootless_tutorial.md]: Added note

**Comment by TomSweeneyRedHat** [comment]
Moved to a Google Doc for further edits/comments at:  https://docs.google.com/document/d/1X4XOFTmKVtDQSNxq9NOBzwIPY3QrPpoCs4qmUBoLohM/edit.  If you'd like access, please send me an email and I'll share the doc with you.  Once the edit/reviews are completed in the document, I will move it here for a final review.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat what's the state on this one?

**Comment by TomSweeneyRedHat** [comment]
I've a number of good comments in the doc file that I've got to polish and work with.  Hope to get to it early next week as I'll no longer by the bug czar.

File: docs/tutorials/rootless_tutorial.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  graphroot = "/execdir/myuser/storage"
+  [storage.options]
+    mount_program = "/bin/fuse-overlayfs"
+```
+
+## Enable unprivileged 'ping`
+
+Users running in a non-privileged container may not be able to use the `ping` utility from that container.
+
+If this is requried, the administrator must verify that the UID of the user is part of the range in the `/proc/sys/net/ipv4/ping_group_range` file.
```
Comment (giuseppe) [inline docs/tutorials/rootless_tutorial.md]: s/requried/required/

**Comment by TomSweeneyRedHat** [comment]
I've repushed with the latest comments from the doc reviews included.  I think this is good enough now to merge and then pick at nits afterward as appropriate.

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**